src/sql/azdata.proposed.d.ts

```diff
@@ -763,4 +763,12 @@ declare module 'azdata' {
 		 */
 		headerFilter?: boolean,
 	}
+
+	export interface TextColumnOption {
+		/**
+		 * Specifies column that will provide icon css value.
+		 * iconCssClassColumn will not be displayed as a separate column, instead its value will be interpreted as a icon css value for the text column.
+		 */
+		iconCssClassColumn?: string
```

Comment:
extensions are not and shouldn't be aware of the css class names available in ADS core, like other ADS extensibility APIs, you should define an icon here. for example: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/azdata.proposed.d.ts#L285

Reply:
TextColumnOption defines the options for text column, the icon property here should actually be iconField?string, and the actual iconField in the data should be the type i mentioned above.

Reply:
@alanrenmsft While I agree that extensions shouldn't know about specific CSS classes (that's an implementation detail) I do think we at some point should look into exposing common icons in a way similar to how VS Code does it where you just have string identifiers : 

https://code.visualstudio.com/api/references/icons-in-labels
https://github.com/microsoft/azuredatastudio/blob/main/src/vs/vscode.d.ts#L5318
https://github.com/microsoft/azuredatastudio/blob/main/src/vs/vscode.d.ts#L800

(not something to do for this PR though)

Reply:
I have added imageCollection and iconProviderColumn field that provides an index of image collection to display near the text in text column.